packages/qdrant-loader/src/qdrant_loader/core/pipeline/document_pipeline.py

```
@@ -25,6 +46,75 @@ def __init__(
+                    errors=["Batch processing timed out after 10 minutes"],
+                )
+
+            total_duration = time.time() - start_time
+            embedding_duration = time.time() - embedding_start
+
+            logger.debug(
+                f"⏱️ Embedding + Upsert phase took {embedding_duration:.2f} seconds"
+            )
+            logger.info(
+                f"✅ Batch processing completed: {pipeline_result.success_count} chunks, "
+                f"{pipeline_result.error_count} errors in {total_duration:.2f}s"
```

Ruff (0.15.14)

[warning] Logging statement uses f-string

(G004)